.github/workflows/test.yml

```diff
@@ -0,0 +1,232 @@
+        with:
+          modules-to-cache: SqlServer
+
+      - name: Setup PowerShell module cache
+        id: cacher
+        uses: actions/cache@v2
+        with:
+            path: ${{ steps.psmodulecache.outputs.modulepath }}
+            key: ${{ steps.psmodulecache.outputs.keygen }}
+
+      - name: Setup Chocolatey download cache
+        id: chococache
+        uses: actions/cache@v2
+        with:
+          path: C:\Users\runneradmin\AppData\Local\Temp\chocolatey\
+          key: chocolatey-install
+
+      - name: Setup Cargo build cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            C:\Users\runneradmin\.cargo\registry
+            C:\Users\runneradmin\.cargo\git
+            target
+          key: ${{ runner.os }}-cargo
+
+      - name: Install required PowerShell modules
+        if: steps.cacher.outputs.cache-hit != 'true'
+        shell: powershell
+        run: |
+            Set-PSRepository PSGallery -InstallationPolicy Trusted
+            Install-Module SqlServer -ErrorAction Stop
+
+      - name: Install SQL Server 2019
+        shell: powershell
+        run: |
+          choco install sql-server-2019
+
+      - name: Setup SQL Server 2019
+        shell: powershell
+        run: |
+          Import-Module 'sqlps'
+
+          [reflection.assembly]::LoadWithPartialName("Microsoft.SqlServer.Smo") | Out-Null
+          [reflection.assembly]::LoadWithPartialName("Microsoft.SqlServer.SqlWmiManagement") | Out-Null
+
+          $serverName = $env:COMPUTERNAME
+          $instanceName = "MSSQLSERVER"
+
+          $smo = 'Microsoft.SqlServer.Management.Smo.'
+          $wmi = new-object ($smo + 'Wmi.ManagedComputer')
+          $wmi
+
+          # Enable TCP/IP
+          echo "Enabling TCP/IP"
+          $Tcp = $wmi.GetSmoObject("ManagedComputer[@Name='$serverName']/ServerInstance[@Name='$instanceName']/ServerProtocol[@Name='Tcp']")
+          $Tcp.IsEnabled = $true
+          $Tcp.alter()
+          $Tcp
+
+          # Enable named pipes
+          echo "Enabling named pipes"
+          $Np = $wmi.GetSmoObject("ManagedComputer[@Name='$serverName']/ServerInstance[@Name='$instanceName']/ServerProtocol[@Name='Np']")
+          $Np.IsEnabled = $true
+          $Np.Alter()
+          $Np
+
+          # Set Alias
+          echo "Setting the alias"
+          New-Item HKLM:\SOFTWARE\Microsoft\MSSQLServer\Client -Name ConnectTo | Out-Null
+          Set-ItemProperty -Path HKLM:\SOFTWARE\Microsoft\MSSQLServer\Client\ConnectTo -Name '(local)' -Value "DBMSSOCN,$serverName\$instanceName" | Out-Null
+
+          # Start services
+          echo "Starting services"
+          Set-Service SQLBrowser -StartupType Manual
+          Start-Service SQLBrowser
+          net stop MSSQLSERVER
+          net start MSSQLSERVER
+
+      - name: Run normal tests
+        shell: powershell
+        run: cargo test --features all
+
+  cargo-test-macos:
+    runs-on: macos-latest
+
+    strategy:
+      fail-fast: false
```

Comment:
curious why this is set to false

Reply:
I want to see all the runs in one go. If we fail on 2017, it's useful for me to see if we also fail on 2019.